bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/DecimalValue.java

```diff
@@ -57,19 +62,20 @@ public class DecimalValue implements SimpleValue, BDecimal {
     private final BigDecimal value;
 
     public DecimalValue(BigDecimal value) {
-        this.value = value;
+        this.value = getValidDecimalValue(value);
         if (!this.booleanValue()) {
             this.valueKind = DecimalValueKind.ZERO;
         }
     }
 
     public DecimalValue(String value) {
         // Check whether the number provided is a hexadecimal value.
+        BigDecimal bd;
         if (isHexValueString(value)) {
-            this.value = hexToDecimalFloatingPointNumber(value);
+            bd = hexToDecimalFloatingPointNumber(value);
```

Comment:
Any idea why this line not covered?

Reply:
It looks like previously no tests with hex string have been added to cover the lines of the method `hexToDecimalFloatingPointNumber`. Will try to add some cases for that.

Reply:
According to [spec](https://ballerina.io/spec/lang/2022R2/#lang.decimal) `decimal:String()` is not going to accept hexadecimal strings as arguments and it has been addressed with the issue https://github.com/ballerina-platform/ballerina-lang/issues/33406. It looks like JsonParser also need not to be read and convert hexadecimal strings to decimal values. So the method added to convert hexadecimal string to BigDecimal value in `DecimalValue.java` can be safely removed.